eng/common/pipelines/templates/steps/verify-resource-ref.yml

```diff
@@ -0,0 +1,32 @@
+
+steps:
+  - pwsh: |
+      Install-Module -Name powershell-yaml -RequiredVersion 0.4.1 -Force -Scope CurrentUser
+      $remoteHEAD = git rev-parse origin/master
+      $localHEAD = git rev-parse HEAD
+      $ymlfilesChanged = (git diff --name-only $remoteHEAD...$localHEAD) | Where-Object {$_ -like '*.yml'}
+      
+      foreach ($file in $ymlfilesChanged)
+      {
+        Write-Host "Verifying "$file
+        $ymlContent = Get-Content "$(Build.SourcesDirectory)/$file" -Raw
+        $ymlObject = ConvertFrom-Yaml $ymlContent -Ordered
+
+        if ($ymlObject.Contains("resources"))
+        {
+          if ($ymlObject["resources"].Contains("repositories"))
+          {
+            $repositories = $ymlObject["resources"]["repositories"]
+            foreach ($repo in $repositories)
+            {
+              $repoName = $repo["repository"]
+              if (-not ($repo.Contains("ref")))
+              {
+                Write-Error "Ref not found. Please ensure you add a Ref: when using repository resource"
```

Comment:
I propose that you maintain a list of repos that don't have a ref, rather than exiting just on the first one. This way folks can have an error that is the full summary of ALL their issues.

Reply:
Thanks will do that.

Reply:
I would also add a little more information in the error message explaining why a ref is needed.